discord/client.py

```diff
@@ -804,10 +835,16 @@ async def start(self, token: str, *, reconnect: bool = True) -> None:
         TypeError
             An unexpected keyword argument was received.
         """
+        # Update the loop to get the running one in case the one set is MISSING
         await self.login(token)
         await self.connect(reconnect=reconnect)
 
-    def run(self, *args: Any, **kwargs: Any) -> None:
+    def run(
+        self,
+        token: str,
+        *,
+        reconnect: bool = True,
+    ) -> None:
```

Comment:
This can be a breaking change for some, e.g. people that still pass `bot=True`. But I think it's a good change overall.